WooCommerce/Classes/AppDelegate.swift

```diff
@@ -179,6 +176,9 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // If the application was previously in the background, optionally refresh the user interface.
 
         RequirementsChecker.checkMinimumWooVersionForDefaultStore()
+
+        // Cache onboarding state to speed IPP process
+        refreshCardPresentPaymentsOnboardingIfNeeded()
```

Comment:
An unexpected implication of putting this here: it's called after every TTP transaction.

Resigning active (and therefore becoming active when the app comes back) is a very temporary state, it apparently happens for phone calls, SMS (maybe outdated) and other things where the app isn't in the background, it's just not what's actively receiving events. 

I wonder if it happens even more on iPad with multitasking... I've not tested that.

[Apple's docs]
(https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1622956-applicationdidbecomeactive)

Anyway, one place where we resign active: when Apple's TTP present card screen is shown! So this is refreshing onboarding after every TTP payment, which probably isn't desirable.

[Maybe `applicationWillEnterForeground` is best](https://developer.apple.com/documentation/uikit/uiapplicationdelegate/1623076-applicationwillenterforeground)? It may need to be duplicated in `willFinishLaunching` too.

Reply:
The problem with calling it in `applicationWillEnterForeground`is that it is [not called](https://stackoverflow.com/a/9860393/428353) on the first launch after `willFinishLaunchingWithOptions`. As I see it we have these options as places where we call to refresh:

-  `willFinishLaunchingWithOptions`: Original approach, changed as it might not be called often enough.
- `applicationDidBecomeActive`: Current approach, it might be called too often.
- `applicationWillEnterForeground`: Proposed approach, not called on the first launch.

I personally lean towards the second option, but it's true that being called after the Apple's TTP present card screen is shown is a disadvantage. What do you think?

Reply:
Given these, I think it's best to call it in both `applicationWillEnterForeground` and `willFinishLaunchingWithOptions`, as that should get us exactly what we want – on first launch, and every time we foreground the app. Thanks for checking this, I'd forgotten that we don't get the foreground call for first launch but the moment you wrote it, I remembered other apps where I've had to make the same calls twice. 

The `applicationDidBecomeActive` refreshes the onboarding state after every TTP payment, which is unintended and way too often (and... kind of at the wrong time! If we're going to call it every time, surely it should be just before each payment, not just after!)

Reply:
That's a good combination indeed, done in d9424c1145e1a000082572936962b29f923538cd